### PR TITLE
Refacto

### PR DESCRIPTION
**Pull request by raefko**
Removed some deadcode.
Cleaned the code.
Add a good documentation (an HTML can be build with sphinx).
Handle get_code type of contract.
Add a test to check if the disassembler crash while parsing/disassembling ...

### CODE REVIEW

File: tests/test.py [hunk 1]
```diff
@@ -1,6 +1,8 @@
+from lib2to3.pytree import Base
```
Comment (pventuzelo) [inline tests/test.py:1]: what is it? some stuff added by vscode?

File: thoth/abi_parser.py [hunk 2]
```diff
@@ -142,27 +187,28 @@ def extract_functions(json_type, json_data):
...
         func_identifiers = extract_function_prototype(
             func_offset, identifiers_data, entry_points_by_type
         )
-
     else:
-        print("Sorry, json retrieve using `get_code` is not supported yet")
-        print("Please consider using `get_full_contract` instead")
-        sys.exit(1)
-
+        func_offset["0"] = "unkown_function"
```
Comment (pventuzelo) [inline thoth/abi_parser.py:197]: "unknown" and not "unkown"

File: thoth/cfg.py [hunk 3]
```diff
@@ -118,8 +133,6 @@ def _generate_basicblocks(self, instructions):
                 new_bb = True
 
             else:
-                # Should never happen
-                # all function finish by RET
```
Comment (pventuzelo) [inline thoth/cfg.py:122]: why removing those comments?

File: tests/test.py (hunk 1 above)
Comment (pventuzelo) [inline tests/test.py:1]: what is it? added by vscode?

File: thoth/disassembler.py [hunk 4]
```diff
@@ -121,34 +150,28 @@ def print_disassembly(self, func_name=None, func_offset=None):
         print("_" * 75)
 
         if self.functions == []:
-            print()
-            print("No bytecode/functions found (maybe it's a contract interface?)")
```
Comment (pventuzelo) [inline thoth/disassembler.py:125]: maybe add a FAQ in the readme with this case

**Comment by pventuzelo** [comment]
Please also add inside the FAQ how to generate the HTML doc

File: thoth/cfg.py (hunk 3 above)
Comment (raefko) [reply thoth/cfg.py:122]: i moved them to the documentation 
`        """Generate the internal list of BasicBlock
        Args:
            instructions (List): List of instructions

        Raises:
            AssertionError: Should never happen, all function finish by RET
        """`

File: thoth/abi_parser.py (hunk 2 above)
Comment (raefko) [reply thoth/abi_parser.py:197]: Oupsi, fixed

File: tests/test.py (hunk 1 above)
Comment (raefko) [reply tests/test.py:1]: Yes, sorry..

**Review by pventuzelo — APPROVED** [review]

**Pull request merged**